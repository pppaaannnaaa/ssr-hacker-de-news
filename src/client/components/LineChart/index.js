import React from 'react';
import PropTypes from 'prop-types';
import { Chart } from 'react-google-charts';

const LineChart = (props) => {
  const { data } = props;
  const lineChart = [['x', 'Vote'], ...data];
  const options = {
    title: 'News de Hacker Line Chart',
    hAxis: {
      title: 'ID',
    },
    vAxis: {
      title: 'Votes',
    },
  };
  const loaderNode = (
    <div style={{ padding: '200px 20%' }}>
      <div className="progress">
        <div className="indeterminate" />
      </div>
    </div>
  );

  return (
    <div role="form" aria-label="Line Chart">
      <Chart
        width="100%"
        height="500px"
        chartType="LineChart"
        loader={loaderNode}
        data={lineChart}
        options={options}
        rootProps={{ 'data-testid': '1' }}
      />
    </div>
  );
};

LineChart.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  data: PropTypes.arrayOf(PropTypes.any),
};

LineChart.defaultProps = {
  data: [],
};

export default LineChart;
