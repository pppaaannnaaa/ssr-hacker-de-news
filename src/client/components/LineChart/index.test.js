import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LineChart from './index';

describe('<LineChart />', () => {
  let props;

  beforeEach(() => {
    props = {
      loaderNode: <div />,
      options: {
        title: 'News de Hacker Line Chart',
        hAxis: {
          title: 'ID',
        },
        vAxis: {
          title: 'Votes',
        },
      },
      lineChart: [
        ['x', 'Vote'],
        [1, 2],
        [2, 4],
        [3, 6],
        [4, 8],
      ],
    };
  });

  afterEach(cleanup);

  test('Renders <LineChart /> component correctly', async () => {
    const { container } = render(
      <LineChart loader={props.loaderNode} data={props.lineChart} options={props.options} />
    );
    expect(container).toMatchSnapshot();
  });
});
