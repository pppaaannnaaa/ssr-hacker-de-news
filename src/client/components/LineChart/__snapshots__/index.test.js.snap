// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChart /> Renders <LineChart /> component correctly 1`] = `
<div>
  <div
    aria-label="Line Chart"
    role="form"
  >
    <div
      style="padding: 200px 20%;"
    >
      <div
        class="progress"
      >
        <div
          class="indeterminate"
        />
      </div>
    </div>
  </div>
</div>
`;
